ampd/src/queue/queued_broadcaster.rs

```diff
@@ -81,23 +70,17 @@ where
         batch_gas_limit: Gas,
         capacity: usize,
         broadcast_interval: Duration,
-    ) -> (Self, QueuedBroadcasterDriver) {
-        let (broadcast_tx, broadcast_rx) = mpsc::channel(1);
-
-        (
-            Self {
-                broadcaster,
-                queue: MsgQueue::default(),
-                batch_gas_limit,
-                broadcast_interval,
-                channel: mpsc::channel(capacity),
-                broadcast_rx,
-            },
-            QueuedBroadcasterDriver { broadcast_tx },
-        )
+    ) -> Self {
+        Self {
+            broadcaster,
+            queue: MsgQueue::default(),
+            batch_gas_limit,
+            broadcast_interval,
+            channel: mpsc::channel(capacity),
+        }
     }
 
-    pub async fn run(mut self) -> Result {
+    pub async fn run(self) -> Result {
         let (tx, mut rx) = self.channel;
         drop(tx);
```

Comment:
```suggestion
        let (_, mut rx) = self.channel;
```
is this equivalent?

Reply:
Unfortunately, it is not. The unit tests hang with it. I've tried.